src/game/behaviors/spawn_star.inc.c

```diff
@@ -123,30 +123,37 @@ void bhv_star_spawn_loop(void) {
     }
 }
 
-struct Object *spawn_star(struct Object *starObj, f32 x, f32 y, f32 z) {
-    starObj = spawn_object_abs_with_rot(o, 0, MODEL_STAR, bhvStarSpawnCoordinates, o->oPosX, o->oPosY, o->oPosZ, 0, 0, 0);
+struct Object *spawn_star(f32 x, f32 y, f32 z) {
+    struct Object *starObj = spawn_object_abs_with_rot(o, 0, MODEL_STAR, bhvStarSpawnCoordinates, o->oPosX, o->oPosY, o->oPosZ, 0, 0, 0);
     starObj->oBehParams = o->oBehParams;
     vec3f_set(&starObj->oHomeVec, x, y, z);
     starObj->oFaceAnglePitch = 0;
     starObj->oFaceAngleRoll = 0;
     return starObj;
 }
 
-void spawn_default_star(f32 x, f32 y, f32 z) {
-    struct Object *starObj = NULL;
-    starObj = spawn_star(starObj, x, y, z);
+void spawn_default_star(void) {
+    f32 dist;
+    struct Object *starObj;
+    struct Object *starMarker = cur_obj_find_nearest_object_with_behavior_and_bparams(bhvDefaultStarMarker,
```

Comment:
This is regression compared to vanilla logic where I could specify arbitrary location for a star to spawn. Now if I have multiple spawners, closest will be chosen.

Reply:
This would only be a problem if you have multiple spawners for the same star ID. Otherwise you'd only be able to have one spawner per level (which would be a problem for obvious reasons).

You can always make a new function for this if you have a use case of overriding the spawn marker, but in general hardcoding star coords is worse in almost every other way.

Reply:
If developer has to change code, this PR does nothing good. If you want to go this way, introduce a enum for bparam2 to search for the object based on that bparam2 instead of nearest object.

Reply:
The exact idea here is that they don't. Having to hardcode coordinates for stars means people can't handle that through fast64 alone, and forgetting to change those coords will result in broken spawns almost every time.

All the developer needs to do is add a star marker object via fast64 to set the spawn location, with a matching bparam1. If no star marker is found, it falls back to the object's home as a safeguard (which is still an improvement over hardcoding).

Comparing against bparam2 as an additional identifier isn't a good option here because that bparam2 is inhereted from the parent object (which could be set to anything). In my opinion though the need for this is an edge case that vanilla doesn't do natively anywhere either. If it's _really_ necessary to keep a hardcoded spawn function in HackerSM64, that too can be added to this PR.